engine/src/test/java/org/camunda/bpm/engine/test/dmn/deployment/DecisionDefinitionTest.java

```diff
@@ -0,0 +1,199 @@
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+
+public class DecisionDefinitionTest {
+
+  protected ProcessEngineBootstrapRule bootstrapRule = new ProcessEngineBootstrapRule(configuration -> {
+    configuration.setHistoryTimeToLive("P30D");
+  });
+
+  protected ProvidedProcessEngineRule engineRule = new ProvidedProcessEngineRule(bootstrapRule);
+
+  protected ProcessEngineTestRule testRule = new ProcessEngineTestRule(engineRule);
+
+  @Rule
+  public RuleChain ruleChain = RuleChain.outerRule(bootstrapRule)
+      .around(engineRule)
+      .around(testRule);
+
+  protected RepositoryService repositoryService;
+  protected DecisionService decisionService;
+  protected HistoryService historyService;
+
+  @Before
+  public void init() {
+    this.repositoryService = engineRule.getRepositoryService();
+    this.decisionService = engineRule.getDecisionService();
+    this.historyService = engineRule.getHistoryService();
+  }
+
+  @Test
+  public void shouldUseHistoryTTLOnDecisionDefinitions() {
+    // given
+    DmnModelInstance model = createDmnModelInstance(null);
+
+    DeploymentBuilder builder = repositoryService.createDeployment().addModelInstance("foo.dmn", model);
+
+    // when
+    DeploymentWithDefinitions deployment = testRule.deploy(builder);
+
+    // then
+    assertThat(deployment.getDeployedDecisionDefinitions().size()).isEqualTo(1);
+    assertThat(deployment.getDeployedDecisionDefinitions().get(0).getHistoryTimeToLive()).isEqualTo(30);
+  }
+
+  @Test
+  public void shouldNotOverrideWithGlobalConfigOnDecisionHistoryTTLPresence() {
+    // given
+    DmnModelInstance model = createDmnModelInstance("P10D");
+
+    DeploymentBuilder builder = repositoryService.createDeployment().addModelInstance("foo.dmn", model);
+
+    // when
+    DeploymentWithDefinitions deployment = testRule.deploy(builder);
+
+    // then
+    assertThat(deployment.getDeployedDecisionDefinitions().size()).isEqualTo(1);
+    assertThat(deployment.getDeployedDecisionDefinitions().get(0).getHistoryTimeToLive()).isEqualTo(10);
+  }
+
+  @Test
+  public void shouldApplyHistoryTTLOnRemovalTimeOfDecisionInstanceLocal() {
+    DmnModelInstance model = createDmnModelInstance("P10D");
+    DeploymentBuilder builder = repositoryService.createDeployment().addModelInstance("foo.dmn", model);
+
+    testRule.deploy(builder);
+
+    Map<String, Object> variables = new HashMap<>();
+    variables.put("input", "single entry");
+
+    decisionService.evaluateDecisionByKey("Decision-1")
+        .variables(variables)
+        .evaluate();
+
+    HistoricDecisionInstance result = historyService.createHistoricDecisionInstanceQuery()
+        .singleResult();
+
+    Date expectedRemovalDate = DateTime.now()
+        .plusDays(10)
+        .toDate();
+
+    assertThat(result.getRemovalTime()).isInSameDayAs(expectedRemovalDate);
+  }
+
+  @Test
+  public void shouldApplyHistoryTTLOnRemovalTimeOfDecisionInstanceGlobal() {
```

Comment:
🙃 You could add some more structure with `// given`, `// when`, and `// then` comments.

Reply:
Done.